### PR TITLE
Dynamic generation of terraform properties

### PR DESCRIPTION
**Pull request by sumo-drosiek**
###### Description

Dynamic generation of terraform properties

###### Testing performed

- [ ] ci/build.sh
- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

File: tests/terraform/static/all_fields.output.yaml [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+          name        = "filter.no2"
+          regexp      = ".+"
+        }
+        force_timezone               = true
+        host_name                    = "localhost"
+        manual_prefix_regexp         = ".*"
+        multiline_processing_enabled = true
+        timezone                     = "local timezone"
+        use_autoline_matching        = false
+    }
```
Comment (sumo-drosiek) [inline tests/terraform/static/all_fields.output.yaml:171]: Generating this template is insane :D

**Comment by perk-sumo** [comment]
Other than the tests failing I'm 👍

File: deploy/helm/sumologic/values.yaml [hunk 2]
```diff
@@ -99,6 +99,10 @@ sumologic:
   metrics:
     enabled: true
 
+  ## Configuration of additional collector fields
+  ## https://help.sumologic.com/Manage/Fields#http-source-fields
+  fields: {}
```
Comment (frankreno) [inline deploy/helm/sumologic/values.yaml:104]: Since this is part of setup only, should it leave in the setup section?

Comment (vsinghal13) [reply deploy/helm/sumologic/values.yaml:104]: +1

Comment (perk-sumo) [reply deploy/helm/sumologic/values.yaml:104]: +1

Comment (sumo-drosiek) [reply deploy/helm/sumologic/values.yaml:104]: I coukd, but now `setup` looks like well-structured config, and adding `fields` will break it. I am going to add commit which we can revert eventually

**Pull request merged**